tests/common/devices.py

```diff
@@ -1133,10 +1133,14 @@ def get_extended_minigraph_facts(self, tbinfo):
         # the minigraph facts are not always match up with PTF port indices.
         try:
             dut_index = tbinfo['duts'].index(self.hostname)
-            map = tbinfo['topo']['ptf_map'][str(dut_index)]
-            if map:
+            port_map = tbinfo['topo']['ptf_map'][str(dut_index)]
+
+            if port_map:
                 for port, index in mg_facts['minigraph_port_indices'].items():
-                    mg_facts['minigraph_ptf_indices'][port] = map[str(index)]
+                    try:
+                        mg_facts['minigraph_ptf_indices'][port] = port_map[str(index)]
+                    except (KeyError):
```

Comment:
The issue is because that the DUT switch has more interfaces than actually used interfaces in topology definition. Then accessing `tbinfo['topo']['ptf_map']` based on DUT port index could trigger `KeyError` exception. Suggest to explicitly check if the ptf_map has key of DUT port index. If not, then skip current DUT port. This would be more clear than catching `KeyError`.
BTW, it seems that the outer `try except` is unnecessary.

Reply:
Good suggestion. Have updated and add some comment to make it clear. Thanks.

Reply:
Sorry I missed this PR. I made exact same change to address this issue plus cleaning up cache.

Reply:
My PR is #2978